### PR TITLE
Fix upload large file sizes bug

### LINKED ISSUE #687: 🐛 [Bug] Two Upload Success Messages are sent to the client on Image Upload (User Profile)
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-27T19:48:25Z" title="Monday, May 27th 2019, 9:48:25 pm +02:00">May 27, 2019</time>_
_Closed <time datetime="2019-06-02T02:47:35Z" title="Sunday, June 2nd 2019, 4:47:35 am +02:00">Jun 2, 2019</time>_
---

### Two Upload Success Messages are sent to the client on Image Upload (User Profile)
Two Upload Success Messages are sent to the client on Image Upload (User Profile). Maybe even to uploads take place.

### Steps to reproduce the behavior
1. Upload Userprofile
2. Wait for successful upload
3. See two upload success Messages
4. ...
5. Profit

### Expected behavior

Only one upload and one message should take place

### LINKED ISSUE #691: 🐛 [Bug] Avatar upload - handle large filesizes and error exceeding the limit
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-27T20:47:44Z" title="Monday, May 27th 2019, 10:47:44 pm +02:00">May 27, 2019</time>_
_Closed <time datetime="2019-06-02T02:47:35Z" title="Sunday, June 2nd 2019, 4:47:35 am +02:00">Jun 2, 2019</time>_
---

### Avatar upload - handle large filesizes and error exceeding the limit
Uploading Avatar images over a certain treshhold will result in an error.
The error massage is not good readable, we have to find a different solution for that.
And afterwards one can not upload an image again !!!

- [ ] Define Filesize limit (on backend)
- [ ] Handle File exceeding that limit
- [ ] Display proper error Message (toast?)
- [ ] Make sure functionality is not blocked once an error regarding filesize occured

### Steps to reproduce the behavior
```
3 MB
Bildschirmfoto 2018-03-19 um 10.23.37.png
File is too big (2.83MiB). Max filesize: 0.5MiB.
```

image

### Expected behavior

Should work ;-)

### Additional context

See https://github.com/Human-Connection/Human-Connection/pull/605#pullrequestreview-241640341

### PR DESCRIPTION
**Pull request by Tirokk**
> [aonomike](https://github.com/aonomike) **Authored by [aonomike](https://github.com/aonomike)**
_<time datetime="2019-05-28T20:48:28Z" title="Tuesday, May 28th 2019, 10:48:28 pm +02:00">May 28, 2019</time>_
_Merged <time datetime="2019-06-02T02:47:34Z" title="Sunday, June 2nd 2019, 4:47:34 am +02:00">Jun 2, 2019</time>_
---

## 🍰 Pullrequest

- Increase the minimum accepted image size to 3 MB
- Display toast with error message if maximum size is exceeded
- Resolve double upload submit to the backend

Screen Shot 2019-05-28 at 11 46 10 PM

### Issues

- fixes #691
- fixes #687  

### Todo

- [ ] Force update the image component to display the initial image

### CODE REVIEW

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-29T15:14:20Z" title="Wednesday, May 29th 2019, 5:14:20 pm +02:00">May 29, 2019</time>_

---


this is great @aonomike, almost there! we just need to add tests to cover the new lines of code/functionality

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-06-01T19:24:12Z" title="Saturday, June 1st 2019, 9:24:12 pm +02:00">Jun 1, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/698?src=pr&el=h1) Report
> Merging [#698](https://codecov.io/gh/Human-Connection/Human-Connection/pull/698?src=pr&el=desc) into [master](https://codecov.io/gh/Human-Connection/Human-Connection/commit/13a2f771f86ca96b3fb55de1bd949782edfc608a?src=pr&el=desc) will **increase** coverage by `0.12%`.
> The diff coverage is `83.33%`.

[Impacted file tree graph](https://codecov.io/gh/Human-Connection/Human-Connection/pull/698?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #698      +/-   ##
==========================================
+ Coverage   21.15%   21.27%   +0.12%     
==========================================
  Files         162      162              
  Lines        2274     2280       +6     
  Branches      113      114       +1     
==========================================
+ Hits          481      485       +4     
- Misses       1765     1766       +1     
- Partials       28       29       +1
```


| [Impacted Files](https://codecov.io/gh/Human-Connection/Human-Connection/pull/698?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [webapp/components/Upload/index.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/698/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvVXBsb2FkL2luZGV4LnZ1ZQ==) | `80% <83.33%> (-4.21%)` | :arrow_down: |

File: webapp/components/Upload/index.vue [hunk 1]
```diff
@@ -44,6 +45,14 @@ export default {
       }
     },
   },
+  watch: {
+    error() {
+      let that = this
```
Comment (Tirokk) [inline webapp/components/Upload/index.vue:50]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-03T11:17:12Z" title="Monday, June 3rd 2019, 1:17:12 pm +02:00">Jun 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -44,6 +45,14 @@ export default {
       }
     },
   },
+  watch: {
+    error() {
+      let that = this
```


Was this not solved with EcmaScript? :thinking: could you try out with `error: () => {` arrow functions?

Comment (Tirokk) [reply webapp/components/Upload/index.vue:50]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-03T11:17:59Z" title="Monday, June 3rd 2019, 1:17:59 pm +02:00">Jun 3, 2019</time>_

---


Or, what is `this ` actually here? Could you explain? :thinking:

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-03T11:21:49Z" title="Monday, June 3rd 2019, 1:21:49 pm +02:00">Jun 3, 2019</time>_

---


@aonomike I'm excited to see this in action! Thank you so much!

awesome

File: webapp/components/Upload/index.vue (hunk 1 above)
Comment (Tirokk) [reply webapp/components/Upload/index.vue:50]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-06-04T00:04:58Z" title="Tuesday, June 4th 2019, 2:04:58 am +02:00">Jun 4, 2019</time>_

---


I tried it out and it seems to work, but it breaks the tests, so maybe it doesn't or maybe the tests need to be modified somehow... to be honest, we copied and pasted this from the `vue2-dropzone` demo...

**Review by Mogge — APPROVED** [review]

**Pull request closed without merging**